include/openthread/border_routing.h

```diff
@@ -149,6 +149,22 @@ typedef enum
  */
 otError otBorderRoutingInit(otInstance *aInstance, uint32_t aInfraIfIndex, bool aInfraIfIsRunning);
 
+/**
+ * Changes the Border Routing interface to the given infrastructure interface.
+ *
+ * @param[in]  aInstance          A pointer to an OpenThread instance.
+ * @param[in]  aInfraIfIndex      The infrastructure interface index.
+ * @param[in]  aInfraIfIsRunning  A boolean that indicates whether the infrastructure
+ *                                interface is running.
+ *
+ * @retval  OT_ERROR_NONE           Successfully started the Border Routing Manager on given infrastructure.
+ * @retval  OT_ERROR_INVALID_STATE  The Border Routing Manager has already been initialized.
+ *
+ * @sa otPlatInfraIfStateChanged.
+ *
+ */
+otError otBorderRoutingChangeInfraNetif(otInstance *aInstance, uint32_t aInfraIfIndex, bool aInfraIfIsRunning);
```

Comment:
What about merging `otBorderRoutingInit` and  `otBorderRoutingChangeInfraNetif` to a single  `otBorderRoutingSetInfraNetif` API? @abtink @librasungirl

Reply:
Yes.I agree. 

We can actually keep the existing `otBorderRoutingInit` API name and just update its behavior to allow it to be called again to change it (restrict it to only when BR is stopped).